OneLogin (C2C) (#5645)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/onelogin-source.md

  - cloud-to-cloud
  - onelogin-source
description: Learn how to collect the users list from the OneLogin API and send it to Sumo Logic.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('/img/platform-services/automation-service/app-central/logos/onelogin.png')} alt="logo" width="100" />

OneLogin is a leading identity and access management (IAM) provider that offers secure, centralized access to applications and devices through single sign-on (SSO) and multi-factor authentication (MFA). Its cloud-based platform integrates with a wide range of applications, simplifying IT operations, enhancing security, and supporting compliance. OneLogin also provides tools for provisioning, lifecycle management, and directory synchronization to streamline user administration.

## Data collected

The data will be collected from OneLogin database using the following log:

| Polling Interval | Data |
| :--- | :--- |
| 24 hours | [Users List](https://developers.onelogin.com/api-docs/2/users/list-users) |

## Setup

### Vendor configuration

The OneLogin source requires you to provide the **Account URL**, **Client ID**, and **Client Secret** to set up the integration.

> #### Sub Domain

Reviewer: ```suggestion
#### Subdomain
```

> The Sub Domain is the account URL used to retrieve the source data from the OneLogin API. For example, `https://{subdomain}.onelogin.com`.
> [Learn more](https://developers.onelogin.com/api-docs/2/getting-started/dev-overview) to obtain the sub domain URL.

Reviewer: ```suggestion
The subdomain is the account URL used to retrieve the source data from the OneLogin API. For example, `https://{subdomain}.onelogin.com`. [Learn more](https://developers.onelogin.com/api-docs/2/getting-started/dev-overview) to obtain the subdomain URL.
```

To configure a OneLogin source:
1. [**Classic UI**](/docs/get-started/sumo-logic-ui-classic). In the main Sumo Logic menu, select **Manage Data > Collection > Collection**. <br/>[**New UI**](/docs/get-started/sumo-logic-ui). In the Sumo Logic top menu select **Configuration**, and then under **Data Collection** select **Collection**. You can also click the **Go To...** menu at the top of the screen and select **Collection**.
1. On the Collection page, click **Add Source** next to a Hosted Collector.
1. Search for and select **OneLogin**.
1. Enter a **Name** for the Source. The description is optional.
1. (Optional) For **Source Category**, enter any string to tag the output collected from the source. Category metadata is stored in a searchable field called `_sourceCategory`.
1. (Optional) **Fields**. Click the **+Add** button to define the fields you want to associate. Each field needs a name (key) and value.

>    * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
>    * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema it is ignored, known as dropped.

Reviewer: this template was just updated last night https://github.com/SumoLogic/sumologic-documentation/pull/5634

```suggestion
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists and is enabled in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist, or is disabled in the Fields table schema. In this case, you'll see an option to automatically add or enable the nonexistent fields to the Fields table schema. If a field is sent to Sumo Logic but isn’t present or enabled in the schema, it’s ignored and marked as **Dropped**.
```

> 1. **Sub Domain**. Enter your [sub domain URL](#vendor-configuration).

Reviewer 2: ```suggestion
1. **Sub Domain**. Enter your [subdomain URL](#vendor-configuration).
```

> 1. **Client ID**. Enter your [Client ID](#vendor-configuration).
> 1. **Client Secret**. Enter your [Client Secret](#vendor-configuration).

Reviewer 2: ```suggestion
1. **Client ID**. Enter your [client ID](#vendor-configuration).
1. **Client Secret**. Enter your [client secret](#vendor-configuration).
```
